classes/article/ArticleGalleyDAO.inc.php

```diff
@@ -632,6 +632,25 @@ function deleteAllPubIds($journalId, $pubIdType) {
 		}
 		$this->flushCache();
 	}
+
+	/**
+	 * Delete the public ID of a galley.
+	 * @param $galleyId int
+	 * @param $pubIdType string One of the NLM pub-id-type values or
+	 * 'other::something' if not part of the official NLM list
+	 * (see <http://dtd.nlm.nih.gov/publishing/tag-library/n-4zh0.html>).
+	 */
+	function deletePubId($galleyId, $pubIdType) {
+		$settingName = 'pub-id::'.$pubIdType;
+		$this->update(
+			'DELETE FROM article_galley_settings WHERE setting_name = ? AND galley_id = ?',
+			array(
+				$settingName,
+				(int)$galleyId
+			)
+		);
+		$this->flushCache();
```

Comment:
If you were referring to the fact that ArticleGalleyDAO had no specific implementation of flushCache and relied on what was in DAO, I've added a method override in ArticleGalleyDAO that specifically fetches the ArticleGalley cache and calls $cache->flush() now.  It's part of the new commit from yesterday.